oidc/viewsets.py

```diff
@@ -81,86 +85,103 @@ def logout(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:
             _("Unable to process OpenID connect logout request."),
         )
 
+    def _check_user_exists(self, user_data: dict) -> bool:
+        """
+        Helper function that checks if a user exists
+        """
+        if user_data.get(self.unique_user_filter_field):
+            field_value = user_data.get(self.unique_user_filter_field)
+            field = self.unique_user_filter_field + "__iexact"
+            return self.user_model.objects.filter(**{field: field_value}).count() > 0
+        return True
+
+    def generate_successful_response(self, request, user) -> HttpResponse:
+        """
+        Generates a success response for a successful Open ID Connect
+        Authentication request
+        """
+        response = HttpResponseRedirect(config.get("REDIRECT_AFTER_AUTH"))
+
+        if self.use_auth_backend:
+            login(
+                request, user, backend=self.auth_backend,
+            )
+
+        if self.use_sso:
+            sso_cookie = jwt.encode(
+                {"email": getattr(user, self.sso_cookie, "email")},
+                config.get("JWT_SECRET_KEY"),
+                config.get("JWT_ALGORITHM"),
+            )
+            response.set_cookie(
+                "SSO",
+                value=sso_cookie.decode("utf-8"),
+                max_age=self.cookie_max_age,
+                domain=self.cookie_domain,
+            )
+
+        return response
```

Comment:
Just curious, is there a case we can have `self.use_auth_backend` as `True` and `self.use_sso` as `False`? If there is, will `response` have cookies? I'm asking because I can't seem to see where cookies are set if we aren't using single-sign-on?

Reply:
If `self.use_sso` is `False` no cookie will be set for the response if `use_auth_backend` is `False` too.

If `use_auth_backend` is set to `True` the default backend will set the session cookie for the request via the `login` function.

Reply:
> If use_auth_backend is set to True the default backend will set the session cookie for the request via the login function.

Cool. If `login` function sets the session cookie, does that mean the `response` object that has been defined in line 103 will implicitly have the session cookie?

Reply:
Yes

Reply:
I'm not entirely sure.... usually the request has a session object that is updated by the login function. I'll have to read a bit more into django to confirm